packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp

```diff
@@ -10,20 +10,35 @@ using Azure.ClientGenerator.Core;
 @scenarioService("/azure/client-generator-core/internal")
 namespace _Specs_.Azure.ClientGenerator.Core.Internal;
 
-@doc("This is an internal model.")
-model InternalModel {
+@doc("This is a model.")
+model Model {
```

Comment:
Should we not have 2 different models to test the internal case? Wouldn't Model get generated here because `get` is not internal and then defeat the point of `getInternal`

Reply:
oh I see, this is to test it does actually get generated despite being internal somewhere else

Reply:
Ok so I think we might want some different scenarios here:
- Public model
- Shared model
- Internal model


So i'd say lets organize the scenarios this way:

```tsp
@scenario
op publicOnly(): PublicModel;

@scenario
op internalOnly(): InternalOnly;

// This scenario is both operations
@scenario
interface SharedModels {
  public(): SharedModel;
  internal(): SharedModel;
}
```

and so we have 4 endpoints but the last 2 are under the same scenario

Reply:
(feel free to use different names if that make more sense than public)

Reply:
Good suggestion. `public` is good, easy to understand. I've updated the tsp and mockapi.